pkg/operator/apiserver/controller/workload/workload.go

```diff
@@ -274,6 +289,10 @@ func (c *Controller) updateOperatorStatus(ctx context.Context, previousStatus *o
 		deploymentProgressingCondition.Status = operatorv1.ConditionTrue
 		deploymentProgressingCondition.Reason = "PodsUpdating"
 		deploymentProgressingCondition.Message = fmt.Sprintf("deployment/%s.%s: %d/%d pods have been updated to the latest generation", workload.Name, c.targetNamespace, workload.Status.UpdatedReplicas, desiredReplicas)
+	} else if tooManyMatchingPods {
+		deploymentProgressingCondition.Status = operatorv1.ConditionTrue
+		deploymentProgressingCondition.Reason = "PreviousGenPodsPresent"
+		deploymentProgressingCondition.Message = fmt.Sprintf("deployment/%s.%s: %d pod(s) from the previous generation are still present", workload.Name, c.targetNamespace, len(matchingPods)-int(desiredReplicas))
```

Comment:
This may not always be correct. There may be extra pods for different reasons. E.g. if they are disrupted/evicted for some reason, the deployment controller will create extra pods to account for that.

Reply:
How would you word it, then? Just too many pods?

Reply:
In k8s, this is considered a complete deployment, so it depends on what you want to convey. There can also be extra pods a rollout with maxSurge, but in this case the pods will have different revisions/hashes.

Reply:
Apart from the terminating pods which are a subject of this bug, no.

I guess, there might be some exotic cases where the pods can be owned by another controller, but that can be safely ignored here.